### PR TITLE
own: display error if predicates are used and feature flag is off

### PR DESCRIPTION
**Pull request by leonore**
closes https://github.com/sourcegraph/sourcegraph/issues/48843

Screenshot 2023-03-09 at 10 52 23

## Test plan

Added unit tests and manually tested:
* with no feature flag
* with feature flag on
* with feature flag off

Manually checked that the event_logs table did not log the searches with no fflags/fflag off.

### CODE REVIEW

File: internal/search/job/jobutil/enterprise.go [hunk 1]
```diff
@@ -22,12 +22,12 @@ func NewUnimplementedEnterpriseJobs() EnterpriseJobs {
...
 type enterpriseJobs struct{}
 
-func (e *enterpriseJobs) FileHasOwnerJob(child job.Job, includeOwners, excludeOwners []string) job.Job {
-	return NewUnimplementedJob("file:has.owner searches are not available on this instance")
+func (e *enterpriseJobs) FileHasOwnerJob(child job.Job, features *search.Features, includeOwners, excludeOwners []string) job.Job {
+	return NewUnimplementedJob("`file:has.owner` searches are not available on this instance")
 }
 
-func (e *enterpriseJobs) SelectFileOwnerJob(child job.Job) job.Job {
-	return NewUnimplementedJob("file:select.owners searches are not available on this instance")
```
Comment (eseliger) [inline internal/search/job/jobutil/enterprise.go:30]: lol

File: enterprise/internal/own/search/filter_job.go [hunk 2]
```diff
@@ -16,22 +17,35 @@ import (
 	"github.com/sourcegraph/sourcegraph/lib/errors"
 )
 
-func NewFileHasOwnersJob(child job.Job, includeOwners, excludeOwners []string) job.Job {
+type featureFlagError struct {
+	predicate string
+}
+
+func (e *featureFlagError) Error() string {
+	return fmt.Sprintf("`%s` searches are not enabled on this instance. Enable the `search-ownership` flag to access Ownership search.", e.predicate)
```
Comment (eseliger) [inline enterprise/internal/own/search/filter_job.go:25]: I like that this error is very actionable, but also it's a thing only an admin can do, should we simply link to the docs page or so? or say "an admin can enable it"?

**Review by eseliger — APPROVED** [review]

File: enterprise/internal/own/search/filter_job.go (hunk 2 above)
Comment (limitedmage) [reply enterprise/internal/own/search/filter_job.go:25]: I agree that linking to docs page is the way to go here.

Comment (leonore) [reply enterprise/internal/own/search/filter_job.go:25]: yep, will change this! I couldn't actually find docs, but I think they're not just landed yet. I'll link to https://docs.sourcegraph.com/own?

File: internal/search/job/jobutil/enterprise.go (hunk 1 above)
Comment (leonore) [reply internal/search/job/jobutil/enterprise.go:30]: inception

**Pull request merged**